### PR TITLE
throw 400 when create or update entity with fileNameOverride

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProvider.java [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+		if (EventType.CREATE == event.getType() && entity.getFileNameOverride() != null) {
+			throw new IllegalArgumentException(FILE_NAME_OVERRIDE_DEPRECATED_REASON);
+		}
+	}
+
+	@Override
+	public void entityUpdated(UserInfo userInfo, FileEntity entity) {
+		if (entity.getFileNameOverride() != null) {
+			FileEntity original = (FileEntity) manager.getEntity(userInfo, entity.getId());
+			if (original.getFileNameOverride() != entity.getFileNameOverride()) {
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProvider.java]: need to rewrite the if.

File: services/repository/src/test/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		provider.validateEntity(fileEntity, new EntityEvent(EventType.CREATE, path, userInfo));
+	}
+
+	@Test (expected=IllegalArgumentException.class)
+	public void testUpdateWithoutDataFileHandleId(){
+		provider.validateEntity(fileEntity, new EntityEvent(EventType.UPDATE, path, userInfo));
+	}
+
+	@Test
+	public void testUpdateWithOriginalFileNameOverride(){
```
Comment (john-hill) [inline services/repository/src/test/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProviderTest.java:81]: update test to fail with != check.

File: services/repository/src/test/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		when(mockEntityManager.getEntity(userInfo, fileEntity.getId())).thenReturn(fileEntity);
+		fileEntity.setDataFileHandleId("1");
+		fileEntity.setFileNameOverride("fileNameOverride");
+		provider.entityUpdated(userInfo, fileEntity);
+	}
+
```
Comment (john-hill) [inline services/repository/src/test/java/org/sagebionetworks/repo/web/service/metadata/FileEntityMetadataProviderTest.java:98]: add test for original is null
